packages/firestore/src/remote/persistent_stream.ts

```diff
@@ -201,6 +202,15 @@ export abstract class PersistentStream<
     return this.state === PersistentStreamState.Open;
   }
 
+  /**
+   * Returns true if the underlying RPC can timeout. There are two cases:
+   *   1) The RPC is open and there is no activity;
+   *   2) The RPC failed and we are re-trying connection.
+   */
+  canTimeout(): boolean {
```

Comment:
I have trouble reasoning about this method.  canIdleTimeout() would at least make it clear what timeout we're talking about, though it's still not obvious to me when it would return true...  and the comment doesn't actually match what the code does since isStarted() doesn't necessarily mean "The RPC failed"...  So I think we need to find a better method name and/or rework this code to be more straightforward somehow.